cylc/flow/hostuserutil.py

```diff
@@ -113,24 +114,26 @@ def get_host_ip_by_name(target):
         """Return internal IP address of target."""
         return socket.gethostbyname(target)
 
-    def _get_host_info(self, target=None):
+    def _get_host_info(
+        self, target: Optional[str] = None
+    ) -> Tuple[str, List[str], List[str]]:
         """Return the extended info of the current host."""
+        if target is None:
+            target = socket.getfqdn()
+        if (
+            IS_MAC_OS
+            and target == (
+                '1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.'
+                '0.0.0.0.0.0.ip6.arpa'
+            )
+        ):
+            # Python's socket bindings don't play nicely with mac os
+            # so by default we get the above ip6.arpa address from
+            # socket.getfqdn, note this does *not* match `hostname -f`.
+            # https://github.com/cylc/cylc-flow/issues/2689
+            # https://github.com/cylc/cylc-flow/issues/3595
+            target = socket.gethostname()
         if target not in self._host_exs:
-            if target is None:
-                target = socket.getfqdn()
-            if (
-                IS_MAC_OS
-                and target == (
-                    '1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.'
-                    '0.0.0.0.0.0.ip6.arpa'
-                )
-            ):
```

Comment:
I'm not sure this patch makes sense, `None` should not be in `_host_exs`, if an `arpa` address *is* in `_host_exs`, that's a bug.

What issue were you trying to fix?

Reply:
Whenever `target` is `None` then it assumes the host info is not cached when it might be. Because it goes on to call `socket.getfqdn()`.

It makes more sense to call `socket.getfqdn()` so we have a target in mind, then check if it is cached.

Reply:
So no related issue, just arbitrary "tidying"?